quipucords/tests/integration/test_network_scan.py

```diff
@@ -255,3 +261,114 @@ def test_sanity_check_raw_fact_matches(
         SystemFingerprint.get_valid_fact_names() - unexpected_fingerprints
     )
     assert network_scan_fingerprints == set(fingerprint_fact_map.keys())
+
+
+@pytest.fixture
+def target_ip(faker):
+    """Fake target machine IP used during testing."""
+    return faker.ipv4()
+
+
+@pytest.fixture
+def raw_facts(target_ip):
+    """Return expected raw facts."""
+    _raw_facts = next(raw_facts_generator(DataSources.NETWORK.value, n=1))
+    # HOST_DONE must be true so the scan can be considered successful; connection_host
+    # must match scanned system ip (target_ip)
+    _raw_facts.update(**{HOST_DONE: True, "connection_host": target_ip})
+    return _raw_facts
+
+
+_runner_counter = None
+
+
+@pytest.fixture
+def patched_scan_data(mocker, target_ip, raw_facts):
+    """Patched data for network scan.
+
+    This fixture aims to mock very surgical points on network scan, allowing to exercise
+    most of the code.
+    """
+    global _runner_counter  # noqa: PLW0603
```

Comment:
🤨 

I think I understand your intent here, but seeing `global` _immediately_ sets off my spidey senses. Maybe we can find a slightly less offensive alternative? If you made this a class-based test method, I think you could use a member variable for the counter instead.

Reply:
I can try! Yeah, a class could be a way to get rid of it.